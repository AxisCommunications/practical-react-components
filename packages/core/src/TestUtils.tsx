import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { PracticalProvider } from './Practical'

/*
  Only used for Jest Snapshot Testing
*/

export const TP = ({
  children,
}: {
  readonly children: React.ReactElement
}): JSX.Element => <PracticalProvider>{children}</PracticalProvider>

/**
 * Mock reference to practical-root
 */
const createNodeMock = (element: React.ReactElement) => {
  if (element.props.id !== 'practical-root') {
    return null
  }

  return {}
}

export const TestRender = (jsx: JSX.Element) => {
  return renderer.create(<TP>{jsx}</TP>, { createNodeMock }).toJSON()
}

export const TestText = (): JSX.Element => <span>Test</span>

/* eslint-disable-next-line react/display-name */
export const TestTextWithForwardRef = React.forwardRef<HTMLElement>(
  ({ ...props }, ref) => (
    <span {...props} ref={ref}>
      Test
    </span>
  )
)
