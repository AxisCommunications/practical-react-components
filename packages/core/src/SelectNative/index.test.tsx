import React from 'react'
import 'jest-styled-components'

import { SelectNative } from '.'
import { TestRender } from '../TestUtils'

const options = [
  { value: 'test1', label: 'test1' },
  { value: 'test2', label: 'test2', disabled: true },
  { value: 'test3', label: 'test3' },
]

const callback = () => {
  /** */
}

describe('Draggable', () => {
  test('Width', () => {
    const tree1 = TestRender(
      <SelectNative
        value="test1"
        options={options}
        width="small"
        onChange={callback}
      />
    )
    expect(tree1).toMatchSnapshot()

    const tree2 = TestRender(
      <SelectNative
        value="test1"
        options={options}
        width="medium"
        onChange={callback}
      />
    )
    expect(tree2).toMatchSnapshot()

    const tree3 = TestRender(
      <SelectNative
        value="test1"
        options={options}
        width="large"
        onChange={callback}
      />
    )
    expect(tree3).toMatchSnapshot()

    const tree4 = TestRender(
      <SelectNative
        value="test1"
        options={options}
        width="full"
        onChange={callback}
      />
    )
    expect(tree4).toMatchSnapshot()
  })

  test('Variant', () => {
    const tree1 = TestRender(
      <SelectNative
        value="test1"
        options={options}
        variant="filled"
        onChange={callback}
      />
    )
    expect(tree1).toMatchSnapshot()

    const tree2 = TestRender(
      <SelectNative
        value="test1"
        options={options}
        variant="transparent"
        onChange={callback}
      />
    )
    expect(tree2).toMatchSnapshot()
  })

  test('Disabled', () => {
    const tree1 = TestRender(
      <SelectNative
        value="test1"
        options={options}
        disabled={true}
        onChange={callback}
      />
    )
    expect(tree1).toMatchSnapshot()
  })
})
