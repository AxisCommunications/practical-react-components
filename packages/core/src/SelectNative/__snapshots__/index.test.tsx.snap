// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Draggable Disabled 1`] = `
.c6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  pointer-events: none;
  overflow: hidden;
  vertical-align: middle;
  height: 24px;
  width: 24px;
}

.c6 > svg {
  fill: currentColor;
  height: 100%;
  width: auto;
}

.c2 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  color: rgb(122,122,122);
}

.c5 {
  position: absolute;
  right: 8px;
  background-color: transparent;
}

.c5 > svg {
  width: 100%;
  height: 100%;
  display: inline-block;
}

.c3 {
  height: 40px;
  width: 100%;
  color: rgb(41,41,41);
  background-color: rgb(245,245,245);
  border-width: 0;
  border: 1px solid transparent;
  border-radius: 4px;
  padding: 0 8px;
  cursor: pointer;
  font-family: Open Sans,Helvetica,arial,sans-serif;
  font-size: 13px;
  line-height: 22px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  text-indent: 1px;
  text-overflow: '';
}

.c3:hover {
  background-color: rgb(240,240,240);
}

.c3:focus {
  outline: none;
  background-color: rgb(245,245,245);
  border: 1px solid rgb(61,61,61);
}

.c3:active {
  background-color: rgb(240,240,240);
  border: 1px solid transparent;
}

.c3:disabled {
  opacity: 0.48;
  pointer-events: none;
}

.c3:required:invalid {
  color: rgb(41,41,41);
}

.c4 {
  color: rgb(41,41,41);
  background-color: rgb(255,255,255);
}

.c4:disabled {
  color: rgb(122,122,122);
  pointer-events: none;
}

.c8 {
  position: fixed;
  top: 0;
  width: 360px;
  border-radius: 4px;
  margin: 8px 16px;
  right: 0;
}

.c0 {
  color: rgb(41,41,41);
}

.c1 {
  position: absolute;
  z-index: 0;
}

.c7 {
  position: absolute;
  z-index: 1;
}

<div
  className="c0"
  id="practical-root"
>
  <div>
    <div
      className="c1"
      id="layer-13"
    >
      <div
        className="c2"
        width="full"
      >
        <select
          className="c3"
          disabled={true}
          name=""
          onChange={[Function]}
          required={true}
          value="test1"
        >
          <option
            className="c4"
            value="test1"
          >
            test1
          </option>
          <option
            className="c4"
            disabled={true}
            value="test2"
          >
            test2
          </option>
          <option
            className="c4"
            value="test3"
          >
            test3
          </option>
        </select>
        <div
          className="c5"
        >
          <span
            className="c6"
            role="img"
            size="medium"
          >
            <svg
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M8.71 11.71l2.59 2.59c.39.39 1.02.39 1.41 0l2.59-2.59c.63-.63.18-1.71-.71-1.71H9.41c-.89 0-1.33 1.08-.7 1.71z"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="c7"
      id="layer-14"
    >
      <div
        className="c8"
      />
    </div>
  </div>
</div>
`;

exports[`Draggable Variant 1`] = `
.c6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  pointer-events: none;
  overflow: hidden;
  vertical-align: middle;
  height: 24px;
  width: 24px;
}

.c6 > svg {
  fill: currentColor;
  height: 100%;
  width: auto;
}

.c2 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  color: rgb(122,122,122);
}

.c5 {
  position: absolute;
  right: 8px;
  background-color: transparent;
}

.c5 > svg {
  width: 100%;
  height: 100%;
  display: inline-block;
}

.c3 {
  height: 40px;
  width: 100%;
  color: rgb(41,41,41);
  background-color: rgb(245,245,245);
  border-width: 0;
  border: 1px solid transparent;
  border-radius: 4px;
  padding: 0 8px;
  cursor: pointer;
  font-family: Open Sans,Helvetica,arial,sans-serif;
  font-size: 13px;
  line-height: 22px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  text-indent: 1px;
  text-overflow: '';
}

.c3:hover {
  background-color: rgb(240,240,240);
}

.c3:focus {
  outline: none;
  background-color: rgb(245,245,245);
  border: 1px solid rgb(61,61,61);
}

.c3:active {
  background-color: rgb(240,240,240);
  border: 1px solid transparent;
}

.c3:disabled {
  opacity: 0.48;
  pointer-events: none;
}

.c3:required:invalid {
  color: rgb(41,41,41);
}

.c4 {
  color: rgb(41,41,41);
  background-color: rgb(255,255,255);
}

.c4:disabled {
  color: rgb(122,122,122);
  pointer-events: none;
}

.c8 {
  position: fixed;
  top: 0;
  width: 360px;
  border-radius: 4px;
  margin: 8px 16px;
  right: 0;
}

.c0 {
  color: rgb(41,41,41);
}

.c1 {
  position: absolute;
  z-index: 0;
}

.c7 {
  position: absolute;
  z-index: 1;
}

<div
  className="c0"
  id="practical-root"
>
  <div>
    <div
      className="c1"
      id="layer-9"
    >
      <div
        className="c2"
        width="full"
      >
        <select
          className="c3"
          disabled={false}
          name=""
          onChange={[Function]}
          required={true}
          value="test1"
        >
          <option
            className="c4"
            value="test1"
          >
            test1
          </option>
          <option
            className="c4"
            disabled={true}
            value="test2"
          >
            test2
          </option>
          <option
            className="c4"
            value="test3"
          >
            test3
          </option>
        </select>
        <div
          className="c5"
        >
          <span
            className="c6"
            role="img"
            size="medium"
          >
            <svg
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M8.71 11.71l2.59 2.59c.39.39 1.02.39 1.41 0l2.59-2.59c.63-.63.18-1.71-.71-1.71H9.41c-.89 0-1.33 1.08-.7 1.71z"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="c7"
      id="layer-10"
    >
      <div
        className="c8"
      />
    </div>
  </div>
</div>
`;

exports[`Draggable Variant 2`] = `
.c6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  pointer-events: none;
  overflow: hidden;
  vertical-align: middle;
  height: 24px;
  width: 24px;
}

.c6 > svg {
  fill: currentColor;
  height: 100%;
  width: auto;
}

.c2 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  color: rgb(122,122,122);
}

.c5 {
  position: absolute;
  right: 8px;
  background-color: transparent;
}

.c5 > svg {
  width: 100%;
  height: 100%;
  display: inline-block;
}

.c3 {
  height: 40px;
  width: 100%;
  color: rgb(41,41,41);
  background-color: transparent;
  border-width: 0;
  border: 1px solid transparent;
  border-radius: 4px;
  padding: 0 8px;
  cursor: pointer;
  font-family: Open Sans,Helvetica,arial,sans-serif;
  font-size: 13px;
  line-height: 22px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  text-indent: 1px;
  text-overflow: '';
}

.c3:hover {
  background-color: rgb(245,245,245);
}

.c3:focus {
  outline: none;
  background-color: transparent;
  border: 1px solid rgb(61,61,61);
}

.c3:active {
  background-color: rgb(240,240,240);
  border: 1px solid transparent;
}

.c3:disabled {
  opacity: 0.48;
  pointer-events: none;
}

.c3:required:invalid {
  color: rgb(41,41,41);
}

.c4 {
  color: rgb(41,41,41);
  background-color: rgb(255,255,255);
}

.c4:disabled {
  color: rgb(122,122,122);
  pointer-events: none;
}

.c8 {
  position: fixed;
  top: 0;
  width: 360px;
  border-radius: 4px;
  margin: 8px 16px;
  right: 0;
}

.c0 {
  color: rgb(41,41,41);
}

.c1 {
  position: absolute;
  z-index: 0;
}

.c7 {
  position: absolute;
  z-index: 1;
}

<div
  className="c0"
  id="practical-root"
>
  <div>
    <div
      className="c1"
      id="layer-11"
    >
      <div
        className="c2"
        width="full"
      >
        <select
          className="c3"
          disabled={false}
          name=""
          onChange={[Function]}
          required={true}
          value="test1"
        >
          <option
            className="c4"
            value="test1"
          >
            test1
          </option>
          <option
            className="c4"
            disabled={true}
            value="test2"
          >
            test2
          </option>
          <option
            className="c4"
            value="test3"
          >
            test3
          </option>
        </select>
        <div
          className="c5"
        >
          <span
            className="c6"
            role="img"
            size="medium"
          >
            <svg
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M8.71 11.71l2.59 2.59c.39.39 1.02.39 1.41 0l2.59-2.59c.63-.63.18-1.71-.71-1.71H9.41c-.89 0-1.33 1.08-.7 1.71z"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="c7"
      id="layer-12"
    >
      <div
        className="c8"
      />
    </div>
  </div>
</div>
`;

exports[`Draggable Width 1`] = `
.c6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  pointer-events: none;
  overflow: hidden;
  vertical-align: middle;
  height: 24px;
  width: 24px;
}

.c6 > svg {
  fill: currentColor;
  height: 100%;
  width: auto;
}

.c2 {
  width: 160px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  color: rgb(122,122,122);
}

.c5 {
  position: absolute;
  right: 8px;
  background-color: transparent;
}

.c5 > svg {
  width: 100%;
  height: 100%;
  display: inline-block;
}

.c3 {
  height: 40px;
  width: 100%;
  color: rgb(41,41,41);
  background-color: rgb(245,245,245);
  border-width: 0;
  border: 1px solid transparent;
  border-radius: 4px;
  padding: 0 8px;
  cursor: pointer;
  font-family: Open Sans,Helvetica,arial,sans-serif;
  font-size: 13px;
  line-height: 22px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  text-indent: 1px;
  text-overflow: '';
}

.c3:hover {
  background-color: rgb(240,240,240);
}

.c3:focus {
  outline: none;
  background-color: rgb(245,245,245);
  border: 1px solid rgb(61,61,61);
}

.c3:active {
  background-color: rgb(240,240,240);
  border: 1px solid transparent;
}

.c3:disabled {
  opacity: 0.48;
  pointer-events: none;
}

.c3:required:invalid {
  color: rgb(41,41,41);
}

.c4 {
  color: rgb(41,41,41);
  background-color: rgb(255,255,255);
}

.c4:disabled {
  color: rgb(122,122,122);
  pointer-events: none;
}

.c8 {
  position: fixed;
  top: 0;
  width: 360px;
  border-radius: 4px;
  margin: 8px 16px;
  right: 0;
}

.c0 {
  color: rgb(41,41,41);
}

.c1 {
  position: absolute;
  z-index: 0;
}

.c7 {
  position: absolute;
  z-index: 1;
}

<div
  className="c0"
  id="practical-root"
>
  <div>
    <div
      className="c1"
      id="layer-1"
    >
      <div
        className="c2"
        width="small"
      >
        <select
          className="c3"
          disabled={false}
          name=""
          onChange={[Function]}
          required={true}
          value="test1"
        >
          <option
            className="c4"
            value="test1"
          >
            test1
          </option>
          <option
            className="c4"
            disabled={true}
            value="test2"
          >
            test2
          </option>
          <option
            className="c4"
            value="test3"
          >
            test3
          </option>
        </select>
        <div
          className="c5"
        >
          <span
            className="c6"
            role="img"
            size="medium"
          >
            <svg
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M8.71 11.71l2.59 2.59c.39.39 1.02.39 1.41 0l2.59-2.59c.63-.63.18-1.71-.71-1.71H9.41c-.89 0-1.33 1.08-.7 1.71z"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="c7"
      id="layer-2"
    >
      <div
        className="c8"
      />
    </div>
  </div>
</div>
`;

exports[`Draggable Width 2`] = `
.c6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  pointer-events: none;
  overflow: hidden;
  vertical-align: middle;
  height: 24px;
  width: 24px;
}

.c6 > svg {
  fill: currentColor;
  height: 100%;
  width: auto;
}

.c2 {
  width: 200px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  color: rgb(122,122,122);
}

.c5 {
  position: absolute;
  right: 8px;
  background-color: transparent;
}

.c5 > svg {
  width: 100%;
  height: 100%;
  display: inline-block;
}

.c3 {
  height: 40px;
  width: 100%;
  color: rgb(41,41,41);
  background-color: rgb(245,245,245);
  border-width: 0;
  border: 1px solid transparent;
  border-radius: 4px;
  padding: 0 8px;
  cursor: pointer;
  font-family: Open Sans,Helvetica,arial,sans-serif;
  font-size: 13px;
  line-height: 22px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  text-indent: 1px;
  text-overflow: '';
}

.c3:hover {
  background-color: rgb(240,240,240);
}

.c3:focus {
  outline: none;
  background-color: rgb(245,245,245);
  border: 1px solid rgb(61,61,61);
}

.c3:active {
  background-color: rgb(240,240,240);
  border: 1px solid transparent;
}

.c3:disabled {
  opacity: 0.48;
  pointer-events: none;
}

.c3:required:invalid {
  color: rgb(41,41,41);
}

.c4 {
  color: rgb(41,41,41);
  background-color: rgb(255,255,255);
}

.c4:disabled {
  color: rgb(122,122,122);
  pointer-events: none;
}

.c8 {
  position: fixed;
  top: 0;
  width: 360px;
  border-radius: 4px;
  margin: 8px 16px;
  right: 0;
}

.c0 {
  color: rgb(41,41,41);
}

.c1 {
  position: absolute;
  z-index: 0;
}

.c7 {
  position: absolute;
  z-index: 1;
}

<div
  className="c0"
  id="practical-root"
>
  <div>
    <div
      className="c1"
      id="layer-3"
    >
      <div
        className="c2"
        width="medium"
      >
        <select
          className="c3"
          disabled={false}
          name=""
          onChange={[Function]}
          required={true}
          value="test1"
        >
          <option
            className="c4"
            value="test1"
          >
            test1
          </option>
          <option
            className="c4"
            disabled={true}
            value="test2"
          >
            test2
          </option>
          <option
            className="c4"
            value="test3"
          >
            test3
          </option>
        </select>
        <div
          className="c5"
        >
          <span
            className="c6"
            role="img"
            size="medium"
          >
            <svg
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M8.71 11.71l2.59 2.59c.39.39 1.02.39 1.41 0l2.59-2.59c.63-.63.18-1.71-.71-1.71H9.41c-.89 0-1.33 1.08-.7 1.71z"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="c7"
      id="layer-4"
    >
      <div
        className="c8"
      />
    </div>
  </div>
</div>
`;

exports[`Draggable Width 3`] = `
.c6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  pointer-events: none;
  overflow: hidden;
  vertical-align: middle;
  height: 24px;
  width: 24px;
}

.c6 > svg {
  fill: currentColor;
  height: 100%;
  width: auto;
}

.c2 {
  width: 300px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  color: rgb(122,122,122);
}

.c5 {
  position: absolute;
  right: 8px;
  background-color: transparent;
}

.c5 > svg {
  width: 100%;
  height: 100%;
  display: inline-block;
}

.c3 {
  height: 40px;
  width: 100%;
  color: rgb(41,41,41);
  background-color: rgb(245,245,245);
  border-width: 0;
  border: 1px solid transparent;
  border-radius: 4px;
  padding: 0 8px;
  cursor: pointer;
  font-family: Open Sans,Helvetica,arial,sans-serif;
  font-size: 13px;
  line-height: 22px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  text-indent: 1px;
  text-overflow: '';
}

.c3:hover {
  background-color: rgb(240,240,240);
}

.c3:focus {
  outline: none;
  background-color: rgb(245,245,245);
  border: 1px solid rgb(61,61,61);
}

.c3:active {
  background-color: rgb(240,240,240);
  border: 1px solid transparent;
}

.c3:disabled {
  opacity: 0.48;
  pointer-events: none;
}

.c3:required:invalid {
  color: rgb(41,41,41);
}

.c4 {
  color: rgb(41,41,41);
  background-color: rgb(255,255,255);
}

.c4:disabled {
  color: rgb(122,122,122);
  pointer-events: none;
}

.c8 {
  position: fixed;
  top: 0;
  width: 360px;
  border-radius: 4px;
  margin: 8px 16px;
  right: 0;
}

.c0 {
  color: rgb(41,41,41);
}

.c1 {
  position: absolute;
  z-index: 0;
}

.c7 {
  position: absolute;
  z-index: 1;
}

<div
  className="c0"
  id="practical-root"
>
  <div>
    <div
      className="c1"
      id="layer-5"
    >
      <div
        className="c2"
        width="large"
      >
        <select
          className="c3"
          disabled={false}
          name=""
          onChange={[Function]}
          required={true}
          value="test1"
        >
          <option
            className="c4"
            value="test1"
          >
            test1
          </option>
          <option
            className="c4"
            disabled={true}
            value="test2"
          >
            test2
          </option>
          <option
            className="c4"
            value="test3"
          >
            test3
          </option>
        </select>
        <div
          className="c5"
        >
          <span
            className="c6"
            role="img"
            size="medium"
          >
            <svg
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M8.71 11.71l2.59 2.59c.39.39 1.02.39 1.41 0l2.59-2.59c.63-.63.18-1.71-.71-1.71H9.41c-.89 0-1.33 1.08-.7 1.71z"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="c7"
      id="layer-6"
    >
      <div
        className="c8"
      />
    </div>
  </div>
</div>
`;

exports[`Draggable Width 4`] = `
.c6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  pointer-events: none;
  overflow: hidden;
  vertical-align: middle;
  height: 24px;
  width: 24px;
}

.c6 > svg {
  fill: currentColor;
  height: 100%;
  width: auto;
}

.c2 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  color: rgb(122,122,122);
}

.c5 {
  position: absolute;
  right: 8px;
  background-color: transparent;
}

.c5 > svg {
  width: 100%;
  height: 100%;
  display: inline-block;
}

.c3 {
  height: 40px;
  width: 100%;
  color: rgb(41,41,41);
  background-color: rgb(245,245,245);
  border-width: 0;
  border: 1px solid transparent;
  border-radius: 4px;
  padding: 0 8px;
  cursor: pointer;
  font-family: Open Sans,Helvetica,arial,sans-serif;
  font-size: 13px;
  line-height: 22px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  text-indent: 1px;
  text-overflow: '';
}

.c3:hover {
  background-color: rgb(240,240,240);
}

.c3:focus {
  outline: none;
  background-color: rgb(245,245,245);
  border: 1px solid rgb(61,61,61);
}

.c3:active {
  background-color: rgb(240,240,240);
  border: 1px solid transparent;
}

.c3:disabled {
  opacity: 0.48;
  pointer-events: none;
}

.c3:required:invalid {
  color: rgb(41,41,41);
}

.c4 {
  color: rgb(41,41,41);
  background-color: rgb(255,255,255);
}

.c4:disabled {
  color: rgb(122,122,122);
  pointer-events: none;
}

.c8 {
  position: fixed;
  top: 0;
  width: 360px;
  border-radius: 4px;
  margin: 8px 16px;
  right: 0;
}

.c0 {
  color: rgb(41,41,41);
}

.c1 {
  position: absolute;
  z-index: 0;
}

.c7 {
  position: absolute;
  z-index: 1;
}

<div
  className="c0"
  id="practical-root"
>
  <div>
    <div
      className="c1"
      id="layer-7"
    >
      <div
        className="c2"
        width="full"
      >
        <select
          className="c3"
          disabled={false}
          name=""
          onChange={[Function]}
          required={true}
          value="test1"
        >
          <option
            className="c4"
            value="test1"
          >
            test1
          </option>
          <option
            className="c4"
            disabled={true}
            value="test2"
          >
            test2
          </option>
          <option
            className="c4"
            value="test3"
          >
            test3
          </option>
        </select>
        <div
          className="c5"
        >
          <span
            className="c6"
            role="img"
            size="medium"
          >
            <svg
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M8.71 11.71l2.59 2.59c.39.39 1.02.39 1.41 0l2.59-2.59c.63-.63.18-1.71-.71-1.71H9.41c-.89 0-1.33 1.08-.7 1.71z"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="c7"
      id="layer-8"
    >
      <div
        className="c8"
      />
    </div>
  </div>
</div>
`;
